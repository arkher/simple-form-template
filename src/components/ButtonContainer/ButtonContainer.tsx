import React, { HTMLAttributes } from "react";

interface IButtonContainer extends HTMLAttributes<HTMLDivElement> {
  message?: string;
  error?: boolean;
}

const ButtonContainer = ({ children, error }: IButtonContainer) => {
  return (
    <div
      className="w-full flex flex-1 flex-row items-center justify-end md:mt-16"
      data-testid="button-container"
    >
      {error !== undefined && error && (
        <div className="text-sm absolute bottom-20 left-8 right-8 w-100 flex flex-1 justify-center text-red-500 md:relative md:bottom-0 md:right-0 md:left-0 md:mr-8 md:justify-end">
          Falha ao enviar resultado. Tente novamente.
        </div>
      )}
      {error !== undefined && !error && (
        <div className="text-sm absolute bottom-20 left-8 right-8 w-100 flex flex-1 justify-center text-green-500 md:relative md:bottom-0 md:right-0 md:left-0 md:mr-8 md:justify-end">
          Resultado enviado com sucesso!
        </div>
      )}
      <div className="flex justify-end">{children}</div>
    </div>
  );
};

export default ButtonContainer;
