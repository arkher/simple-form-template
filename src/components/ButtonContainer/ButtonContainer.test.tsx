// ButtonContainer
import React from "react";
import { render, screen } from "@testing-library/react";
import ButtonContainer from "./ButtonContainer";

describe("ButtonContainer", () => {
  it("should render on the screen", () => {
    render(<ButtonContainer data-testid="button-container" error />);
    expect(screen.findByTestId("button-container")).toBeTruthy();
  });

  it("should render the error message when error prop is true", () => {
    render(<ButtonContainer error />);
    expect(
      screen.getByText("Falha ao enviar resultado. Tente novamente.")
    ).toBeTruthy();
  });

  it("should render the success message when error prop is false", () => {
    render(<ButtonContainer error={false} />);
    expect(screen.getByText("Resultado enviado com sucesso!")).toBeTruthy();
  });

  it("should match snapshot when error is true", () => {
    render(<ButtonContainer error />);
    expect(screen.getByTestId("button-container")).toMatchSnapshot();
  });

  it("should match snapshot when error is FALSE", () => {
    render(<ButtonContainer error={false} />);
    expect(screen.getByTestId("button-container")).toMatchSnapshot();
  });
});
