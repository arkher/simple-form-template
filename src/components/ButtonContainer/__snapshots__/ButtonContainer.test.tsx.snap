// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should match snapshot when error is FALSE 1`] = `
<div
  class="w-full flex flex-1 flex-row items-center justify-end md:mt-16"
  data-testid="button-container"
>
  <div
    class="text-sm absolute bottom-20 left-8 right-8 w-100 flex flex-1 justify-center text-green-500 md:relative md:bottom-0 md:right-0 md:left-0 md:mr-8 md:justify-end"
  >
    Resultado enviado com sucesso!
  </div>
  <div
    class="flex justify-end"
  />
</div>
`;

exports[`ButtonContainer should match snapshot when error is true 1`] = `
<div
  class="w-full flex flex-1 flex-row items-center justify-end md:mt-16"
  data-testid="button-container"
>
  <div
    class="text-sm absolute bottom-20 left-8 right-8 w-100 flex flex-1 justify-center text-red-500 md:relative md:bottom-0 md:right-0 md:left-0 md:mr-8 md:justify-end"
  >
    Falha ao enviar resultado. Tente novamente.
  </div>
  <div
    class="flex justify-end"
  />
</div>
`;
